Tests/PubNubTests/Networking/Routers/HistoryRouterTests.swift

```diff
@@ -570,8 +615,200 @@ extension MessageHistoryEndpointTests {
   }
 }
 
-// MARK: - Response Decoder
+// MARK: - Message Counts
+
+extension HistoryRouterTests {
+  func testMessageCounts_Router() {
+    let router = HistoryRouter(.messageCounts(channels: testSingleChannel, timetoken: 0, channelsTimetoken: [0]),
+                               configuration: config)
+
+    XCTAssertEqual(router.endpoint.description, "Message Counts")
+    XCTAssertEqual(router.category, "Message Counts")
+    XCTAssertEqual(router.service, .history)
+  }
+
+  func testMessageCounts_Router_ValidationError() {
+    let router = HistoryRouter(.messageCounts(channels: [], timetoken: 0, channelsTimetoken: [0]),
+                               configuration: config)
+    XCTAssertEqual(router.validationError?.pubNubError?.details.first, ErrorDescription.emptyChannelArray)
+
+    let missingTTs = HistoryRouter(.messageCounts(channels: testSingleChannel, timetoken: nil, channelsTimetoken: nil),
+                                   configuration: config)
+    XCTAssertEqual(missingTTs.validationError?.pubNubError?.details.first, ErrorDescription.missingTimetoken)
+
+    let invalidTTConfig = HistoryRouter(.messageCounts(channels: testMultiChannels, timetoken: 0, channelsTimetoken: [0]),
+                                        configuration: config)
+    XCTAssertEqual(invalidTTConfig.validationError?.pubNubError?.details.first, ErrorDescription.invalidHistoryTimetokens)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line should be 120 characters or less: currently 122 characters](https://app.codacy.com/app/PubNub/swift/pullRequest?prid=4513555)